public/javascripts/ng-apps.js

```diff
@@ -1,17 +1,86 @@
-var springInitiative = angular.module('springInitiative', ['ngRoute']);
+var springInitiative = angular.module('springInitiativeApp', ['ui.router']);
 
-springInitiative.config(function($routeProvider) {
-  $routeProvider
-    .when('/student', {
-      templateUrl: 'views/student.html',
-      controller: 'mainController'
-    })
-    .when('/login', {
-      templateUrl: 'views/login.html',
+springInitiative.config(function($stateProvider, $urlRouterProvider) {
+
+  $urlRouterProvider.otherwise('/login');
+
+  $stateProvider
+
+    .state('login', {
+      url: '/login',
+      templateUrl: 'views/partial-login.html',
       controller: 'loginCtrl'
     })
-    .when('/', {
-      templateUrl: 'views/login.html',
-      controller: 'loginCtrl'
+
+    .state('index', {
+      url: '/',
+      views: {
+        '': {
+            templateUrl: 'views/index.html',
```

Comment:
Ok cool, could you explain the difference between them now then? I thought index was like the the template file for the ngroutes, but what does main do now?